Refactor room background color logic in map data API

JoJohowl opened the pull request:
Simplifies and updates the logic for determining room background color. Now, rooms with occupants are set to white (#FFFFFF) and vacant rooms to green (#B7F527), removing previous checks for student/contact colors and regex-based format validation.

## Change Summary


## Other Information

[Any other information]

close #58

JoJohowl committed:
Refactor room background color logic in map data API

Simplifies and updates the logic for determining room background color. Now, rooms with occupants are set to white (#FFFFFF) and vacant rooms to green (#B7F527), removing previous checks for student/contact colors and regex-based format validation.

leonleerl reviewed (changes requested):
It seems the logic is not correct
<img width="1512" height="909" alt="Image" src="https://github.com/user-attachments/assets/27803d91-fb32-4038-8454-94307a5af183" />

JoJohowl committed:
Filter empty summaries and update vacant room color

Now only non-empty summaries are added to roomMap for students and contacts. Also, the background color for vacant rooms has been changed from #B7F527 to #459C07.
